### DISCUSSION TITLE
Issues with Upgrading Terraform

### DISCUSSION THREAD

**Discussion by lucabartmann** [kube-hetzner/terraform-hcloud-kube-hetzner#1533] [Q&A]
Hey Guys,

Today i upgraded terraform via `terraform init --upgrade` to get the updated cluster-autoscaler container images, since the old ones are getting removed by Hetzner.

After this i ran `terraform apply` and everything worked besides from one job. This got me the following error:

```
module.kube-hetzner.null_resource.kustomization (remote-exec): The ClusterRoleBinding "system-upgrade" is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:"rbac.authorization.k8s.io", Kind:"ClusterRole", Name:"system-upgrade-controller"}: cannot change roleRef
╷
│ Error: remote-exec provisioner error
│ 
│   with module.kube-hetzner.null_resource.kustomization,
│   on .terraform/modules/kube-hetzner/init.tf line 327, in resource "null_resource" "kustomization":
│  327:   provisioner "remote-exec" {
│ 
│ error executing "/tmp/terraform_168394776.sh": Process exited with status 1
```

Can somebody help me?

**Comment by PatrickRogg** [comment]
I'm having the same issue. Anyone able to help?

Edit: What worked for me is to just delete the `clusterolebinding` and then running terraform apply again

```
kubectl delete clusterrolebinding system-upgrade
```

**Reply by jr-dimedis** [reply]
@PatrickRogg: Thanks for this information.

For me it was necessary to remove the finalizer from the `clusterrolebinding` as well:

```
kubectl patch clusterrolebinding system-upgrade -p '{"metadata":{"finalizers":[]}}' --type=merge
```
